src/spanned.rs

```diff
@@ -0,0 +1,137 @@
+//! ```
+//! #[macro_use]
+//! extern crate serde_derive;
+//!
+//! extern crate toml;
+//! use toml::Spanned;
+//!
+//! #[derive(Deserialize)]
+//! struct Value {
+//!     s: Spanned<String>,
+//! }
+//!
+//! fn main() {
+//!     let t = "s = \"value\"\n";
+//!
+//!     let u: Value = toml::from_str(t).unwrap();
+//!
+//!     assert_eq!(u.s.start(), 4);
+//!     assert_eq!(u.s.end(), 11);
+//!     assert_eq!(u.s.as_ref(), "value");
+//!     assert_eq!(u.s.take(), String::from("value"));
+//! }
+//! ```
+
+use serde::{de, ser};
+use std::convert;
+use std::fmt;
+
+#[doc(hidden)]
+pub const NAME: &'static str = "$__toml_private_Spanned";
+#[doc(hidden)]
+pub const START: &'static str = "$__toml_private_start";
+#[doc(hidden)]
+pub const END: &'static str = "$__toml_private_end";
+#[doc(hidden)]
+pub const VALUE: &'static str = "$__toml_private_value";
+
+/// A spanned value, indicating the range at which it is defined in the source.
+#[derive(Debug)]
+pub struct Spanned<T> {
+    /// The start range.
+    start: usize,
+    /// The end range (exclusive).
+    end: usize,
+    /// The spanned value.
+    value: T,
+}
+
+impl<T> Spanned<T> {
+    /// Access the start of the span.
+    pub fn start(&self) -> usize {
+        self.start
+    }
+
+    /// Access the end of the span.
+    pub fn end(&self) -> usize {
+        self.end
+    }
+
+    /// Access the span for the spanned value.
+    pub fn span(&self) -> (usize, usize) {
+        (self.start, self.end)
+    }
+
+    /// Take the inner value.
+    pub fn take(self) -> T {
```

Comment:
For consistency with libstd, mind naming this `into_inner`? I think we may want to go ahead and add methods like `get_ref` and `get_mut` here as well

Reply:
Done, thanks!